lib/PreviewProvider.js

```diff
@@ -238,35 +197,18 @@ PreviewProvider.prototype = {
       // Add tippy top data, if available
       link = this._tippyTopProvider.processSite(link);
 
-      if (link.cache_key && ss.storage.embedlyData[link.cache_key]) {
-        ss.storage.embedlyData[link.cache_key].accessTime = now;
-        return Object.assign({}, ss.storage.embedlyData[link.cache_key], link);
+      // Find the item in the map and return it if it exists
+      if (existingLinks.has(link.cache_key)) {
+        return Object.assign({}, existingLinks.get(link.cache_key), link);
       } else {
         return previewsOnly ? null : link;
       }
-    })
-    .filter(link => link);
+    }).filter(link => link);
 
-    // gives the opportunity to have at least one run with an old
-    // preview cache. This is for the case where one hasn't opened
-    // the browser since `cacheTTL`
-    this.cleanUpCacheMaybe();
     this._tabTracker.handlePerformanceEvent(event, "previewCacheHits", results.length);
     this._tabTracker.handlePerformanceEvent(event, "previewCacheMisses", processedLinks.length - results.length);
     return results;
-  },
-
-  /**
-   * Determine if a cached link has expired
-   */
-  _isLinkExpired(link) {
-    const cachedLink = ss.storage.embedlyData[link.cache_key];
-    if (!cachedLink) {
-      return false;
-    }
-    let currentTime = Date.now();
-    return (currentTime - cachedLink.refreshTime) > this.options.cacheRefreshAge;
-  },
+  }),
 
   /**
    * Build the preview cache
```

Comment:
This function never gets called by anyone. Perhaps it's for the PreviewProvider cache :)